evennia/contrib/rpg/rpsystem/rpsystem.py

```diff
@@ -1333,30 +1336,30 @@ def get_search_result(
         """
         # we also want to use the default search method
         search_obj = super().get_search_result
-        is_builder = self.locks.check_lockstring(self, "perm(Builder)")
+        is_builder = self.permissions.check("Builder")
+        results = []
 
-        if candidates:
-            candidates = parse_sdescs_and_recogs(
+        if candidates is not None:
+            searched_results = parse_sdescs_and_recogs(
                 self, candidates, _PREFIX + searchdata, search_mode=True
             )
-            results = []
-            for candidate in candidates:
-                # we search by candidate keys here; this allows full error
-                # management and use of all kwargs - we will use searchdata
-                # in eventual error reporting later (not their keys). Doing
-                # it like this e.g. allows for use of the typeclass kwarg
-                # limiter.
-                results.extend(
-                    [obj for obj in search_obj(candidate.key, **kwargs) if obj not in results]
-                )
-
-            if not results and is_builder:
-                # builders get to do a global search by key+alias
-                results = search_obj(searchdata, **kwargs)
+            if not searched_results and is_builder:
+                # builders get to do a search by key
+                results = search_obj(searchdata, candidates=candidates, **kwargs)
```

Comment:
this search is intended as a fallback when no sdesc/recog matches are found, so it should be done only when there are no sdesc/recog matches